### ISSUE TITLE
Stripping AOT generated binary changes behavior

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/sdk#45197]
Noted online here: https://news.ycombinator.com/item?id=26334931

```
> dart hello.dart
hello
> dart compile exe hello.dart
> ./hello.exe
hello
> strip hello.exe
> ./hello.exe
Usage: dart [<vm-flags>] <dart-script-file> [<script-arguments>]

Executes the Dart script <dart-script-file> with the given list of <script-arguments>.

Common VM flags:
--enable-asserts
  Enable assert statements.
...
--version
  Print the SDK version.
```

Repro'ed on 2.12 stable:
```
Dart SDK version: 2.12.0 (stable) (Thu Feb 25 19:50:53 2021 +0100) on "linux_x64"
```

**Comment by mraleph** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/39106

**Comment by mraleph** [comment]
tldr version is that these binaries are build more or less by concatenating two ELF binaries together so they are not really strippable with `strip` (also there is no real reason to strip them - because there is nothing to remove).

**Comment by Tanishq-Banyal** [comment]
@mraleph then why does stripping reduces file size from 5.3mb to 4.4mb ?

Is there any strip flags which can be used to safely strip ?
